packages/forms/src/model.ts

```diff
@@ -430,27 +428,115 @@ export abstract class AbstractControl {
+    this._composedValidatorFn = coerceToValidator(validators);
+  }
+
+  /**
+   * Sets the asynchronous validators that are active on this control. Calling this
+   * overwrites any existing asynchronous validators.
+   *
+   * When you add or remove a validator at run time, you must call
+   * `updateValueAndValidity()` for the new validation to take effect.
+   *
+   * If you want to add a new validator without affecting existing ones, consider
+   * using `addAsyncValidators` method instead.
+   */
+  setAsyncValidators(validators: AsyncValidatorFn|AsyncValidatorFn[]|null): void {
+    this._rawAsyncValidators = validators;
+    this._composedAsyncValidatorFn = coerceToAsyncValidator(validators);
+  }
+
+  /**
+   * Add a synchronous validator or validators to this control, without affecting other validators.
+   *
+   * When you add or remove a validator at run time, you must call
+   * `updateValueAndValidity()` for the new validation to take effect.
+   *
+   * Adding a validator that already exists will have no effect.
+   *
+   * @param validators The new validator function or functions to add to this control.
    */
-  setValidators(newValidator: ValidatorFn|ValidatorFn[]|null): void {
-    this._rawValidators = newValidator;
-    this._composedValidatorFn = coerceToValidator(newValidator);
+  addValidators(validators: ValidatorFn|ValidatorFn[]): void {
+    this.setValidators(addValidators(validators, this._rawValidators));
   }
 
   /**
-   * Sets the async validators that are active on this control. Calling this
-   * overwrites any existing async validators.
+   * Add an asynchronous validator or validators to this control, without affecting other
+   * validators.
    *
    * When you add or remove a validator at run time, you must call
    * `updateValueAndValidity()` for the new validation to take effect.
    *
+   * Adding a validator that already exists will have no effect.
+   *
+   * @param validators The new asynchronous validator function or functions to add to this control.
+   */
+  addAsyncValidators(validators: AsyncValidatorFn|AsyncValidatorFn[]): void {
+    this.setAsyncValidators(addValidators(validators, this._rawAsyncValidators));
+  }
+
+  /**
+   * Remove a synchronous validator from this control, without affecting other validators.
+   * Validators are compared by function reference; you must pass a reference to the exact same
+   * validator function as the one that was originally set.
+   *
+   * When you add or remove a validator at run time, you must call
+   * `updateValueAndValidity()` for the new validation to take effect.
+   *
+   * @param validators The validator or validators to remove.
+   */
+  removeValidators(validators: ValidatorFn|ValidatorFn[]): void {
+    const validatorsToRemove = makeValidatorsArray(validators);
+    const oldValidators = makeValidatorsArray(this._rawValidators);
+    this.setValidators(oldValidators.filter(v => !validatorsToRemove.includes(v)));
+  }
```

Comment:
The `removeValidators` and `removeAsyncValidators` methods can also benefit from moving common logic outside of the AbstractControl class, similar to "add" methods:
```suggestion
  removeValidators(validators: ValidatorFn|ValidatorFn[]): void {
    this.setValidators(removeValidators(validators, this._rawValidators));
  }
```